src/runtime/hexagon_remote/host_malloc.cpp

```diff
@@ -134,19 +189,84 @@ extern "C" {
 __attribute__((weak)) void remote_register_buf(void *buf, int size, int fd);
 
 void halide_hexagon_host_malloc_init() {
-    if (ion_fd != -1) return;
-    pthread_mutex_init(&allocations_mutex, NULL);
+    if (ion_fd != -1) {
+        return;
+    }
+    if (dmabufAllocator != NULL) {
+        return;
+    }
+    use_libdmabuf = false;
     use_newer_ioctl = false;
-    ion_fd = open("/dev/ion", O_RDONLY, 0);
-    if (ion_fd < 0) {
-        __android_log_print(ANDROID_LOG_ERROR, "halide", "open('/dev/ion') failed");
+    use_libion = false;
+
+    __android_log_print(ANDROID_LOG_INFO, "halide", "entering halide_hexagon_host_malloc_init");
```

Comment:
I really don't think we should do this kind of unconditional logging.

Reply:
Removed.